cg/store/models.py

```diff
@@ -1,25 +1,66 @@
-import datetime as dt
 import re
-
-from sqlalchemy import Column, ForeignKey, Table, UniqueConstraint, orm, types
-from sqlalchemy.orm import declarative_base
+from datetime import datetime
+from typing import Annotated, Optional
+
+from sqlalchemy import (
+    BLOB,
+    DECIMAL,
+    VARCHAR,
+    BigInteger,
+    Column,
+    ForeignKey,
+    Numeric,
+    String,
+    Table,
+    Text,
+    UniqueConstraint,
+    orm,
+    types,
+)
+from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column
 from sqlalchemy.orm.attributes import InstrumentedAttribute
 from sqlalchemy.util import deprecated
 
-from cg.constants import (
-    PREP_CATEGORIES,
-    SEX_OPTIONS,
-    STATUS_OPTIONS,
-    DataDelivery,
-    FlowCellStatus,
-    Priority,
-    Workflow,
-)
+from cg.constants import DataDelivery, FlowCellStatus, Priority, Workflow
 from cg.constants.archiving import PDC_ARCHIVE_LOCATION
-from cg.constants.constants import CONTROL_OPTIONS, CaseActions, PrepCategory
+from cg.constants.constants import (
+    CaseActions,
+    ControlOptions,
+    PrepCategory,
+    SexOptions,
+    StatusOptions,
+)
 from cg.constants.priority import SlurmQos
 
-Model = declarative_base()
+bigint = Annotated[int, None]
+blob = Annotated[bytes, None]
+dec = Annotated[float, None]
+num_6_2 = Annotated[float, 6]
+str32 = Annotated[str, 32]
+str64 = Annotated[str, 64]
+str128 = Annotated[str, 128]
+str255 = Annotated[str, 255]
+str256 = Annotated[str, 256]
+text = Annotated[str, None]
+var128 = Annotated[str, 128]
+
+intpk = Annotated[int, mapped_column(primary_key=True)]
```

Comment:
Can we call this `primary_key` or something similar instead?

Reply:
Sure, we can call it something more verbose. I do think int should be included though since strings can also be primary keys. Is `int_primary_key` too long?

Reply:
discussed here: https://github.com/Clinical-Genomics/cg/pull/2961#discussion_r1494359716